src/android/SplashScreen.java

```diff
@@ -307,12 +311,45 @@ public void run() {
 
                 // Create and show the dialog
                 splashDialog = new Dialog(context, android.R.style.Theme_Translucent_NoTitleBar);
+                splashWindow = splashDialog.getWindow();
                 // check to see if the splash screen should be full screen
                 if ((cordova.getActivity().getWindow().getAttributes().flags & WindowManager.LayoutParams.FLAG_FULLSCREEN)
                         == WindowManager.LayoutParams.FLAG_FULLSCREEN) {
                     splashDialog.getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN,
                             WindowManager.LayoutParams.FLAG_FULLSCREEN);
                 }
+
+                // Inspirated in https://github.com/apache/cordova-plugin-splashscreen/pull/124/files
+                String statusBarColor = preferences.getString("SplashStatusBarBackgroundColor", "#000000");
+
+                if (statusBarColor != null && !statusBarColor.isEmpty() && Build.VERSION.SDK_INT >= 19) {
+
+                    splashWindow.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
+                    splashWindow.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+                    try {
+                        // Using reflection makes sure any 5.0+ device will work without having to compile with SDK level 21
+                        splashWindow.getClass().getDeclaredMethod("setStatusBarColor", int.class).invoke(splashWindow, Color.parseColor(statusBarColor));
+                    } catch (Exception ignore) {
+                        // this should not happen, only in case Android removes this method in a version > 21
+                        LOG.w("SplashScreen StatusBarColor", "Method window.setStatusBarColor not found for SDK level " + Build.VERSION.SDK_INT);
```

Comment:
I've tested this inside a Android emulators and it works fine on API 27, 28 and 29.

P.S. But it doesn't work on devices with EMUI 10 and API v29:
```
SplashScreen StatusBarColor: »»» USING COLOR #ffd51e2b
SplashScreen StatusBarColor: Method window.setStatusBarColor not found for SDK level 29
SplashScreen StatusBarColor: »»» java.lang.NoSuchMethodException: com.android.internal.policy.HwPhoneWindow.setStatusBarColor

Reply:
Given that we’ve dropped support for SDK <=21, we could remove this reflection hack altogether and just call the method normally. The next version of the splashscreen will be a major, too, so no problem to drop Android <=5.0 support here,too.